libkbfs/key_manager.go

```diff
@@ -53,6 +53,21 @@ func (km *KeyManagerStandard) GetTLFCryptKeyForBlockDecryption(
 	return km.getTLFCryptKeyUsingCurrentDevice(ctx, kmd, blockPtr.KeyGen, true)
 }
 
+// GetTLFCryptKeyFromAllGenerations implements the KeyManager interface for
+// KeyManagerStandard.
+func (km *KeyManagerStandard) GetTLFCryptKeyFromAllGenerations(
+	ctx context.Context, kmd KeyMetadata) (keys []TLFCryptKey, err error) {
+	for g := KeyGen(FirstValidKeyGen); g <= kmd.LatestKeyGeneration(); g++ {
+		var key TLFCryptKey
+		key, err = km.getTLFCryptKeyUsingCurrentDevice(ctx, kmd, g, false)
```

Comment:
I _think_ it's safe to set the `cache` parameter to `true` here.  We don't have it set to `true` in `GetTLFCryptKeyForEncryption` because there used to be some danger that `Rekey` would call that method before the MD was finalized.  But when getting all the key generations on a finalized MD, it seems safe to cache them.